src/index.ts

```diff
@@ -133,10 +140,32 @@ export function createSelectorCreator<
       const params = []
       const length = dependencies.length
 
+      const paramsCopy = []
+
       for (let i = 0; i < length; i++) {
         // apply arguments instead of spreading and mutate a local list of params for performance.
         // @ts-ignore
         params.push(dependencies[i].apply(null, arguments))
+
+        if (process.env.NODE_ENV !== 'production' && inputStabilityCheck) {
```

Comment:
:hand: Can we move _all_ the logic inside the main `NODE_ENV` check for simplicity?  I'm fine with copy-pasting the loop.

Reply:
sure, i just figured two loops would be worse for performance than one

Reply:
tests say i was wrong 🎉